### ISSUE TITLE
Error with testcontainers and version 1.2.0

### ISSUE THREAD

**Issue by radovanradic** [mariadb-corporation/mariadb-connector-r2dbc#76]
Hello,

We are using r2dbc test containers for junit tests for MariaDB. There were no issues with url like this `r2dbc:tc:mariadb:///databasename?TC_IMAGE_TAG=latest` in previous version (1.1.4). After upgrading to 1.2.0 we are getting this error

`java.lang.IllegalArgumentException: Wrong argument value '' for HaMode`

Attached is simple reproducing example (just run Main). If you change `implementation("org.mariadb:r2dbc-mariadb:1.2.0")` in `build.gradle.kts` to `implementation("org.mariadb:r2dbc-mariadb:1.1.4")` then app works without errors.

Example to reproduce the error: 
[mariadb-test.zip](https://github.com/mariadb-corporation/mariadb-connector-r2dbc/files/14411247/mariadb-test.zip)

Thanks,
Radovan

**Comment by mdeso** [comment]
This can be fixed by changing line 99 of `MariadbConnectionConfiguration` to
`this.haMode = haMode == null || haMode.equals("") ? HaMode.NONE : HaMode.from(haMode);`
(`|| haMode.equals("")` added)

**Comment by mdeso** [comment]
I have investigated a possible fix for this but noticed that this only occurs if the connection string contains an explicit Parameter `&haMode=` (i.e., with an empty value string). The code in the MariaDB driver for parsing haMode hasn't changed recently and for the URL parsing it uses `io.r2dbc.spi.ConnectionFactoryOptions`. Therefore, I think this error is due to a regression in either Micronaut, Testcontainers, or maybe `ConnectionFactoryOptions`, although the latter would probably have caused problems in other areas too and haMode is specific to the MariaDB driver. I don't think that the driver should consider an empty value in this case as valid.

**Comment by mdeso** [comment]
I was able got get testcontainers working again with MariaDB R2DBC by using protocol: "mariadb:none" in my application-test.yml. This is probably not a driver issue and the issue can be closed.

**Comment by radovanradic** [comment]
Yes, the example I posted in the issue is not using Micronaut dependencies, only testcontainers. Most likely something changed in 1.2.0 mariadb client. Or maybe testcontainers needs to adjust to this change to resolve the issue. The example uses just this connection url from test containers `r2dbc:tc:mariadb:///databasename?TC_IMAGE_TAG=latest`

**Comment by mdeso** [comment]
Ok, I think I found the problem in the change log of 1.2.0. HaMode now supports a list of aliases and in previous versions the HaMode `NONE` had the alias "" but now the alias list is initialized with `new String[0]` which is causing the error. The fix is to change the initialization to `new String[] {""}`. I will provide a PR with that fix.
